packages/spacecat-shared-rum-api-client/src/functions/traffic-analysis.js

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { utils } from '@adobe/rum-distiller';
+import { classifyTraffic } from '../common/traffic.js';
+
+function getUTM(bundle, type) {
+  return bundle.events
+    .find((e) => e.checkpoint === 'utm' && e.source === `utm_${type}`)
+    ?.target || null;
+}
+
+function getCWV(bundle, metric) {
+  const measurements = bundle.events
+    .filter((e) => e.checkpoint === `cwv-${metric}`)
+    .map((e) => e.value);
+
+  return measurements.length > 0 ? Math.max(...measurements) : null;
+}
+
+function containsEngagedScroll(bundle) {
+  return bundle.events
+    .some((e) => (e.checkpoint === 'viewmedia' || e.checkpoint === 'viewblock') && e.timeDelta >= 10000)
+    ? 1 : 0;
+}
+
+function getNotFound(bundle) {
+  return bundle.events
+    .find((e) => e.checkpoint === '404')
+    ?.source || null;
+}
+
+function getReferrer(bundle) {
+  return bundle.events
+    .find((e) => e.checkpoint === 'enter')
+    ?.source || null;
+}
```

Comment:
`document.referrer` also exists in `navigate` checkpoint. Please also check the source of the navigate checkpoint if exists: https://github.com/adobe/helix-rum-enhancer/blob/main/modules/index.js#L173